sunbird-es-utils/src/main/java/org/sunbird/common/ElasticSearchRestHighImpl.java

```diff
@@ -0,0 +1,907 @@
+
+    GetIndexRequest indexRequest =
+        new GetIndexRequest().indices(mappedIndexAndType.get(JsonKey.INDEX));
+    Promise<Boolean> promise = Futures.promise();
+    ActionListener<Boolean> listener =
+        new ActionListener<Boolean>() {
+          @Override
+          public void onResponse(Boolean getResponse) {
+            if (getResponse) {
+              promise.success(getResponse);
+            } else {
+              promise.success(false);
+            }
+          }
+
+          @Override
+          public void onFailure(Exception e) {
+            promise.failure(e);
+            ProjectLogger.log("ElasticSearchRestHighImpl:healthCheck error " + e.getMessage(), e);
+          }
+        };
+    ConnectionManager.getRestClient().indices().existsAsync(indexRequest, listener);
+
+    return promise.future();
+  }
+
+  /**
+   * Method to execute ES query with the limitation of size set to 0 Currently this is a rest call
+   *
+   * @param index ES indexName
+   * @param type ES type
+   * @param rawQuery actual query to be executed
+   * @return ES response for the query
+   */
+  @SuppressWarnings("unchecked")
+  @Override
+  public Response searchMetricsData(String index, String type, String rawQuery) {
+    long startTime = System.currentTimeMillis();
+    ProjectLogger.log(
+        "ElasticSearchRestHighImpl:searchMetricsData method started at ==" + startTime,
+        LoggerEnum.PERF_LOG);
+    String baseUrl = null;
+    if (!StringUtils.isBlank(System.getenv(JsonKey.SUNBIRD_ES_IP))) {
+      String envHost = System.getenv(JsonKey.SUNBIRD_ES_IP);
+      String[] host = envHost.split(",");
+      baseUrl =
+          "http://"
+              + host[0]
+              + ":"
+              + PropertiesCache.getInstance().getProperty(JsonKey.ES_METRICS_PORT);
+    } else {
+      ProjectLogger.log(
+          "ElasticSearchRestHighImpl:searchMerticData, ES URL from Properties file",
+          LoggerEnum.INFO);
+      baseUrl = PropertiesCache.getInstance().getProperty(JsonKey.ES_URL);
+    }
+    Map<String, String> mappedIndexAndType = ElasticSearchHelper.getMappedIndexAndType(index, type);
+    String requestURL =
+        baseUrl
+            + "/"
+            + mappedIndexAndType.get(JsonKey.INDEX)
+            + "/"
+            + mappedIndexAndType.get(JsonKey.TYPE)
+            + "/"
+            + "_search";
+    Map<String, String> headers = new HashMap<>();
+    headers.put("Content-Type", "application/json");
+    Map<String, Object> responseData = new HashMap<>();
+    try {
+      // TODO:Currently this is making a rest call but needs to be modified to make
+      // the call using
+      // ElasticSearch client
+      String responseStr = HttpUtil.sendPostRequest(requestURL, rawQuery, headers);
+      ObjectMapper mapper = new ObjectMapper();
+      responseData = mapper.readValue(responseStr, Map.class);
+    } catch (IOException e) {
+      throw new ProjectCommonException(
```

Comment:
THis two different error message doesnot make sense for end user, so will user only one error message , but will log it properly , so that will get to know why it's failure.

Reply:
Done